activerecord/test/cases/finder_test.rb

```diff
@@ -22,13 +22,13 @@ class FinderTest < ActiveRecord::TestCase
   fixtures :companies, :topics, :entrants, :developers, :developers_projects, :posts, :comments, :accounts, :authors, :author_addresses, :customers, :categories, :categorizations, :cars
 
   def test_find_by_id_with_hash
-    assert_raises(ActiveRecord::StatementInvalid) do
+    assert_nothing_raised do
```

Comment:
I think this `StatementInvalid` is not by design and unexpected behavior.
Originally `find_by` with a hash generates invalid SQL.

```ruby
  # Before
  Post.find_by_id(limit: 1)
  # => SELECT  "posts".* FROM "posts" WHERE "id"."limit" = ? LIMIT ?  [["limit", 1], ["LIMIT", 1]]

  # After
  Post.find_by_id(limit: 1)
  # => SELECT  "posts".* FROM "posts" WHERE "posts"."id" = ? LIMIT ?  [["id", nil], ["LIMIT", 1]]
```

This PR fixes to generate valid SQL and a hash is to be `nil` by type casting.